yt/frontends/gadget/tests/test_outputs.py

```diff
@@ -107,6 +109,46 @@ def test_multifile_read():
     assert isinstance(data_dir_load(snap_33), GadgetDataset)
     assert isinstance(data_dir_load(snap_33_dir), GadgetDataset)
 
+@requires_file(snap_33)
+def test_particle_subselection():
+    """
+    This checks that we correctly subselect from a dataset, first by making
+    sure we get all the particles, then by comparing manual selections against
+    them.
+    """
+    ds = data_dir_load(snap_33)
+    psc = ParticleSelectionComparison(ds)
+    
+    sp1 = ds.sphere("c", (0.1, "unitary"))
+    assert_equal(psc.compare_dobj_selection(sp1) , True)
```

Comment:
Not necessary to change, but this could just be `assert psc.compare_dobj_selection(sp1)`, right?

Reply:
Yes, it could -- I can't remember now why I used this.